src/main/java/sirius/db/jdbc/batch/BatchQuery.java

```diff
@@ -94,8 +94,10 @@ protected void tryCommit(boolean cascade) {
             try {
                 Watch w = Watch.start();
                 stmt.executeBatch();
-                stmt.getConnection().commit();
-                avarage.addValues(batchBacklog, w.elapsedMillis());
+                if (!stmt.getConnection().getAutoCommit()) {
+                    stmt.getConnection().commit();
```

Comment:
shouldn't we handle this witha Capability?

Reply:
(Database.hasCapability(TRANSACTION) OR COMMIT and fill this properly for the CH driver?